packages/safe-tools-client/app/templates/application.hbs

```diff
@@ -46,7 +46,12 @@
 </Boxel::Dashboard>
 
 <NetworkConnectModal
-  @isOpen={{and this.isShowingConnectModal (not this.wallet.isConnected)}}
+  @isOpen={{and (and this.isShowingConnectModal (not this.wallet.isConnected)) (eq this.wallet.unsupportedConnectCache undefined)}}
   @onClose={{set this 'isShowingConnectModal' false}}
   @isConnected={{this.wallet.isConnected}}
-/>
\ No newline at end of file
+/>
+
+<UnsupportedNetworkModal
+  @isOpen={{not (eq this.wallet.unsupportedConnectCache undefined)}}
+  @onClose={{set this.wallet 'unsupportedConnectCache' undefined}}
```

Comment:
Does it mean we keep the modal opened until the user changes the network?

Reply:
No, until user closes the modal. Isn't that the desired solution?

Reply:
Ah I see, sorry I misunderstood your comments.

Reply:
When do we change the value of `this.didCloseUnsupportedNetworkModal` back to false?

Reply:
It is better for us to keep this code since adding `didCloseUnsupportedNetworkModal` makes the code too complicated in these scenarios.

1. User closes the `UnsupportedNetworkModal` then the value of `didCloseUnsupportedNetworkModal` is `true`. After that, the user reconnects again but still uses an unsupported network, the `UnsupportedNetworkModal` won't show because the `didCloseUnsupportedNetworkModal` value is still `true`.

2. If we set back the value of `didCloseUnsupportedNetworkModal` to false using a render modifier or when clicking connect button, then the `UnsupportedNetworkModal` and the `ConnectWalletModal` will be overlapping because this condition is still satisfied `this.wallet.isConnectedToUnsupportedNetwork && !this.didCloseUnsupportedNetworkModal`.